test/integration/puppetlabs/services/master/master_service_test.clj

```diff
@@ -51,6 +53,77 @@
+                    (get "puppet-v3-report-/*/")
+                    .getCount))))
+
+     (let [resp (http-get "/status/v1/services?level=debug")]
+       (is (= 200 (:status resp)))
+       (let [status (json/parse-string (:body resp) true)]
+         (is (= #{:jruby-metrics :master :status-service}
+                (set (keys status))))
+
+         (is (= 1 (get-in status [:jruby-metrics :service_status_version])))
+         (is (= "running" (get-in status [:jruby-metrics :state])))
+         (is (nil? (schema/check jruby-metrics-core/JRubyMetricsStatusV1
+                                 (get-in status [:jruby-metrics :status]))))
+
+         (is (= jruby-metrics-core/jruby-pool-lock-not-in-use
+                (get-in status [:jruby-metrics :status :experimental
+                                :jruby-pool-lock-status :current-state])))
+
+         (is (= 1 (get-in status [:master :service_status_version])))
+         (is (= "running" (get-in status [:master :state])))
+         (is (nil? (schema/check master-core/MasterStatusV1
+                                 (get-in status [:master :status]))))
+         (testing "HTTP metrics in status endpoint are sorted in order of aggregate amount of time spent"
+           (let [hit-routes #{"total" "puppet-v3-node-/*/"
+                              "puppet-v3-catalog-/*/" "other"}
+                 http-metrics (get-in status [:master :status :experimental :http-metrics])]
+             (testing "'total' should come first since it is the sum of the other endpoints"
+               (is (= "total" (:route-id (first http-metrics)))))
+             (testing "The other two routes that actually received requests should come next"
+               (is (= #{"puppet-v3-node-/*/" "puppet-v3-catalog-/*/"}
+                      (set (map :route-id (rest (take 3 http-metrics)))))))
+             (testing "The aggregate times should be in descending order"
+               (let [aggregate-times (map :aggregate http-metrics)]
+                 (= aggregate-times (reverse (sort aggregate-times)))))
+             (testing "The counts should be accurate for the endpoints that we hit"
+               (let [find-route (fn [route-metrics route-id]
+                                  (first (filter #(= (:route-id %) route-id) route-metrics)))]
+                 (is (= 3 (:count (find-route http-metrics "total"))))
+                 (is (= 1 (:count (find-route http-metrics "puppet-v3-node-/*/"))))
+                 (is (= 1 (:count (find-route http-metrics "puppet-v3-catalog-/*/"))))
+                 (is (= 1 (:count (find-route http-metrics "other"))))))
+             (testing "The counts should be zero for endpoints that we didn't hit"
+               (is (every? #(= 0 %) (map :count
+                                         (filter
+                                          #(not (hit-routes (:route-id %)))
+                                          http-metrics)))))))))
+
```

Comment:
Same comment here about this being a really long test and maybe it would be good to split it up.